### DISCUSSION TITLE
Custom rollupOptions (vite -> build -> rollupOptions) are overridden by default in nuxt

### DISCUSSION THREAD

**Discussion by ArturKrasinski** [nuxt/nuxt#17612] [General]
**Need**:
I need to set the vite build configuration, specifically the value of chunkFileNames and entryFileNames on output in rollupOptions
![image](https://user-images.githubusercontent.com/5808873/168012163-9d762e76-c2a0-4eb7-9b3c-271c7d77d74a.png)

(I need remove hash from file names generated in vite build process)

**Problem**:
Custom configuration dosent work

**Reason**:
[buildClient function](https://github.com/nuxt/framework/blob/cad4edd5a396c447ad99f4016bc856e197c9d85b/packages/vite/src/client.ts#L17) get custom config and overrides it by default config from nuxt ([where chunkFileNames and entryFileNames are seted](https://github.com/nuxt/framework/blob/cad4edd5a396c447ad99f4016bc856e197c9d85b/packages/vite/src/client.ts#L33))

At the moment the only solution is to use the custom hook nuxt and overwrite the configuration again:

**Solution**:
function buildClient should properly use vite.mergeConfig 
and pass the default configuration to it as the first argument and the custom configuration should be passed as the second argument, which will allow you to overwrite the default configuration

_Is there any reason that custom configuration is being overwritten by the default values? However, do I correctly think this is a mistake?_

**Comment by dselivanovvv** [comment]
any updates? have this issue too
